### ISSUE TITLE
Log queries to stderr in tests not working

### ISSUE THREAD

**Issue by pedro-stanaka** [cakephp/cakephp#7606]
I'm having some serious problems to run my tests of an application made with Cake3. First, I've stumbled on the problem that Fixtures does not have support to **Boolean** data type (even though when they come baked they come as boolean :confused:).
Then, when trying to insert my fixtures in PostgreSQL 9.3 it returned the following error (in cli):

    Exception: Unable to insert fixtures for "App\Test\TestCase\Controller\SchoolsControllerTest" test case. 
    SQLSTATE[25P02]: In failed sql transaction: 7 ERROR:  current transaction is aborted, commands ignored until end of transaction block in [/var/www/repos/Projects/carol-projs/busca-esc-php/vendor/cakephp/cakephp/src/TestSuite/Fixture/FixtureManager.php, line 265]

As it can be seen the error is not much self-explanatory, so I've tried to search for a way to debug the SQL being executed. I found on the docs about the parameter 'log' in the connection and enabled it, but when this exception is thrown nothing gets printed on my console window.

Going further I went to check the method ```FixtureManager::_runOperation()```, and I found the following piece of code:

```php

            $db = ConnectionManager::get($connection, false);
            $logQueries = $db->logQueries();
            if ($logQueries) {
                $db->logQueries(false);
            }

```

correct me if I am wrong, but isn't this disabling query log if it is **ENABLED** in the connection config?  Just to test, after the ```if``` I enabled query logging again with: ```$db->logQueries(true);``` and I was able to see the SQL being executed during the test phase.

P.S. I also configured the Log to capture the scope queriesLog.

```php
Log::config('queries', [
    'className' => 'Console',
    'stream' => 'php://stderr',
    'scopes' => ['queriesLog']
]);
```

My Datasource config for sake of completeness:

```php
'test' => [
            'className' => 'Cake\Database\Connection',
            'driver' => 'Cake\Database\Driver\Postgres',
            'persistent' => false,
            'host' => 'localhost',
            'username' => 'school_app',
            'password' => '*******',
            'database' => 'test',
            'encoding' => 'utf8',
            'timezone' => 'UTC',
            'cacheMetadata' => true,
            'quoteIdentifiers' => false,
            'log' => true,
        ],
```

The problem is present in the release 3.1.3 of CakePHP.

**Comment by HavokInspiration** [comment]
Can you provide us the Fixtures that are causing the error ?

**Comment by lorenzo** [comment]
The log is disabled for fixture-generation queries as they are too numerous and repetitive to help actual debugging of test cases. Unfortunately, this means that debugging of problems in fixtures becomes more difficult

**Comment by AD7six** [comment]
Related https://github.com/cakephp/cakephp/pull/5273

There was talk of allowing verbose debugging, but it wasn't implemented. If action is taken for this ticket I would suggest going in that direction.

**Comment by markstory** [comment]
What does the problematic fixture look like?

**Comment by markstory** [comment]
I wonder if this is a similar issue to #7608

**Comment by pedro-stanaka** [comment]
I actually discovered the cause of the error. When I ran the migrations with the verbose output (making changes directly in the core) I saw that the FixtureManager was not dropping the schema, but It was trying to create my constraints again.

 When the transaction reached that point it rolled back because the constraints already existed. My fixture is the following: [SchoolsFixture.php](https://gist.github.com/pedro-stanaka/d873459a08217823a27e)

**Comment by markstory** [comment]
Why weren't the tables being dropped? Usually the fixtures would drop and re-create the constraints. I am interested as to why the constraints weren't being dropped before being reapplied.

**Comment by pedro-stanaka** [comment]
@markstory Now that I dropped everything and started again I can't seem to reproduce the error again. I suspect that for some internal pg motives (like trying to insert a number where the field is boolean) the fixture stopped in the middle and then when I tried to re-run it I got the error. 

Sorry to bother, but what really saved me was the output to the terminal to investigate the cause of the problem.
